libs/common/src/platform/state/implementations/default-active-user-state.ts

```diff
@@ -47,7 +47,7 @@ export class DefaultActiveUserState<T> implements ActiveUserState<T> {
       // We only care about the UserId but we do want to know about no user as well.
       map((a) => a?.id),
       // To avoid going to storage when we don't need to, only get updates when there is a true change.
-      distinctUntilChanged(),
+      distinctUntilChanged((a, b) => a == b), // Treat null and undefined as equal
```

Comment:
This is a looser equality operation that you claim. If the only concern is null/undefined, it needs to be something like 

```suggestion
      distinctUntilChanged((a, b) => a == null || b == null ? a == b : a === b), // Treat null and undefined as equal
```

which falls back to the rxjs default compare if no nulls or undefineds exist